File: go/beautiful-string/beautiful-string.go
package main

import(
	"fmt"
)

func isBeautifulString(inputString string) bool{
	m := make(map[rune]int)
	for _, val := range inputString{
		m[val]++
	}
	
	for key := range m{
		if m[key] < m[key+1]{
			return false
		}
	}
	return true
}

func main(){
	inputString := "bbbaacdafe"

	fmt.Printf("The string %s is beautiful: %v", inputString, isBeautifulString(inputString))
}

File: js/beautiful-string.js
function isBeautifulString(inputString){
    const lettersCharCode = Array.from(Array(26)).map((e, i) => i + 97)
    const alphabet = lettersCharCode.map((x) => String.fromCharCode(x))
    const letters = Array(alphabet.length).fill(0)
    
    for (let letter of inputString) {
        letters[alphabet.indexOf(letter)]++
    }

    
    for (let i = 0; i < letters.length; i++) {
        if (letters[i + 1] > letters[i]) return false
    }
    return true

}

let inputString = "bbbaacdafe"

console.log(`The string "${inputString}" is beautiful:`, isBeautifulString(inputString))